### ISSUE TITLE
Why is using the title attribute a Sufficient Technique for 2.4.4 Link Purpose?

### ISSUE THREAD

**Issue by LaurenceRLewis** [w3c/wcag#1063]
Under the heading 'Intent' on the Understanding document for 2.4.4 Link Purpose (In Context):

`Assistive technology has the ability to provide users with a list of links that are on the Web page. Link text that is as meaningful as possible will aid users who want to choose from this list of links.`

A Sufficient Technique for 2.4.4 Link Purpose (In Context)

`Providing a supplemental description of the purpose of a link using one of the following techniques:
H33: Supplementing link text with the title attribute`

The title attribute text is NOT included in the 'links list' dialog for NVDA or JAWS. However adding descriptive title text to a link is listed as a Sufficient Technique.

Links to pages
2.4.4 Link Purpose (In Context)
[https://www.w3.org/WAI/WCAG21/Understanding/link-purpose-in-context.html](url)

H33: Supplementing link text with the title attribute
[https://www.w3.org/WAI/WCAG21/Techniques/html/H33](url)

**Comment by mraccess77** [comment]
When a link has an accessible name the title attribute becomes the accessible description.  JAWS does have settings to communicate the title including the keystroke insert+tab. It's my personal opinion that screen readers should show the accessible descriptions in the list of links dialog and that bugs be logged against the screen readers for omitting this information.

**Comment by LaurenceRLewis** [comment]
That makes sense, I agree that screen readers should include descriptive title text in the link list dialog (except where it is a duplicate of the link text ).

**Comment by mraccess77** [comment]
You're welcome to keep the bug open for discussion.   I suspect that this technique also remains in WCAG because before ARIA there was no other way to specify additional information and many sites relied on this for a way to provide additional description.   I think most will agree that today the title attribute has limited use since it's not keyboard accessible in all browsers.   But from a screen reader standpoint there are keystrokes to access it.  Not all sufficient techniques are best practice though and others are more preferable than some.

**Comment by JAWS-test** [comment]
SC 2.4.4 is not about the link text itself, but about the context of the link. The context of a link can never be displayed in the link list because a screen reader cannot know what the link context is: 
* same list entry
* same sentence
* same paragraph
* previous heading etc.

In this respect, the title attribute is no better or worse than the other context information. The only difference is that I get the other context information by reading it, while the title attribute is only available when navigating to the link.

**Comment by JAWS-test** [comment]
See: https://github.com/FreedomScientific/VFO-standards-support/issues/374

**Comment by LaurenceRLewis** [comment]
`SC 2.4.4 is not about the link text itself, but about the context of the link. `

I have to disagree on this occasion 😬.  The link text is a part of, or all of the context. 

`The purpose of each link can be determined from the link text alone or from...`

**Comment by mraccess77** [comment]
@JAWS-test if The author has explicitly set the context into the accessible description why wouldn’t the screen reader show it in the links list as the author has explicitly set the context into the accessible description why wouldn’t the screen reader show it in the links list as an option.  I agree it should be an option.  Perhaps after the name.

**Comment by JAWS-test** [comment]
> The link text is a part of, or all of the context

Yeah, I misspoke. What I wanted to say: in my opinion H33 is about the link context and not about the link labeling. In the example there is a link labeling. The title-attribute gives additional hints.

There is also a clear warning at H33: 

> Because of the extensive user agent limitations in supporting access to the title attribute, authors should use caution in applying this technique. For this reason, it is preferred that the author use technique C7 ...

**Comment by LaurenceRLewis** [comment]
I would appreciate feedback on another potential WCAG fail for 2.4.4 Link Purpose.

If an icon / image / svg is wrapped in a link `<a href="somplace.html" title="some place"><graphic with no accessible name></a>`

Ignoring that the image has no accessible name, the fact that keyboard only users cannot access the title attribute, is this a fail.

I ask as I have this exact situation and although I am sure it fails, I would appreciate confirmation, or not.

Thank you.

**Comment by JAWS-test** [comment]
> Ignoring that the image has no accessible name

if "graphic with no accessible name" means that it has an empty alternative text (alt="") or role=presentation it is no problem as long as the link has a label.

> the fact that keyboard only users cannot access the title attribute, is this a fail.

The fact that tooltips are not displayed in most browsers is an error of the browser (see https://www.w3.org/TR/2015/NOTE-UAAG20-20151215/#sc_211) and not an error of the web developer. However, since most browsers do not display the tooltips via the title attribute, one could argue that title is currently not accessibility supported. I would not consider it a problem if the graphics are so well known / meaningful that the purpose of the link is understandable even without a title. In your example, the title is primarily used as an accessible name for the link (for people who cannot see the graphic)

**Comment by mbgower** [comment]
First, let's agree this is not a great construct. As described, it would help no one, especially if the image is also useless (which you don't discuss).

Having done that, I'll address a few points:
> Ignoring that the image has no accessible name

The computation for accessible name includes the value of the title attribute, so I believe this image has an accessible name. If "Some place" was a useful string of text, this would arguably pass Link Purpose. Note that I haven't constructed that and verified, but that's my recollection of the computation.

> the fact that keyboard only users cannot access the title attribute, is this a fail.
Assuming the image conveys the purpose, a keyboard user is going to be able to infer the meaning from the image, right? If the image is meaningless, then the keyboard user is out of luck -- as is a user of the site with a mobile browser.

In a site that is designed with some level of competency, an image link's purpose should be understood by most sighted users. Likewise on a site with some level of content competency, a well written title value will assist those who can use a mouse in better understanding its purpose. And for those who cannot see the image, the title value provides the accessible name.

I don't call this scenario ideal, but I also wouldn't call it a failure of 2.4.4

**Comment by LaurenceRLewis** [comment]
Thanks for the feedback @JAWS-test  and @mbgower 

To clarify this particular situation the image/s are SVG icons, of which for some a guess could be had of the intent, others could have a number of meanings, which is unclear. These icons are wrapped in hyperlinks. The SVG <svg ...>< /svg > do not have an accessible name. 

The < a > includes a title attribute describing the link. This is the only means of understanding the link (in context).

The link has an accName because < title > is a part of the Accessible Name Computation (which I have a reasonable understanding of thanks to Bryan Garaventa's mentoring on the subject). 

Therefore pointing device (mouse) users and screen reader users should be able to discern the purpose of the linked icon. However keyboard only users who do not use assistive technology will need to guess the purpose of the link or click and discover.

Because this group of users may have difficulty understanding the purpose of the link from the icon alone, I would assume this a fail of  2.4.4 Link Purpose (in context). However from the comments I gather this is not the case, and I have to question that.

IMO :-)
> an image link's purpose should be understood by most sighted users.

'Should' is not definitive

> In a site that is designed with some level of competency,

From my experience this is subjective

Thanks again for the feedback.

**Comment by mraccess77** [comment]
It's ironic that changing the the title to an aria-label would provide the same experience to a screen reader user and would then remove the tooltip that may not be keyboard accessible and would then confirm as long as the aria-label described the link purpose.

**Comment by LaurenceRLewis** [comment]
Unfortunately no one solution fits all. It would be fine if title tooltips were keyboard accessible.
